pkg/clients/tekton/bundles.go

```diff
@@ -18,27 +22,32 @@ type Bundles struct {
 // NewBundles returns new Bundles.
 func (t *TektonController) NewBundles() (*Bundles, error) {
 	namespacedName := types.NamespacedName{
-		Name:      "build-pipeline-selector",
+		Name:      "build-pipeline-config",
 		Namespace: "build-service",
 	}
 	bundles := &Bundles{}
-	pipelineSelector := &buildservice.BuildPipelineSelector{}
-	err := t.KubeRest().Get(context.Background(), namespacedName, pipelineSelector)
+	configMap := &corev1.ConfigMap{}
+	err := t.KubeRest().Get(context.Background(), namespacedName, configMap)
 	if err != nil {
 		return nil, err
 	}
-	for i := range pipelineSelector.Spec.Selectors {
-		selector := &pipelineSelector.Spec.Selectors[i]
-		bundleName, bundleRef := tekton.GetPipelineNameAndBundleRef(&selector.PipelineRef)
-		switch bundleName {
+
+	bpc := &tekton.BuildPipelineConfig{}
+	if err = yaml.Unmarshal([]byte(configMap.Data["config.yaml"]), bpc); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal build pipeline config: %v", err)
+	}
+
+	for i := range bpc.Pipelines {
+		pipeline := bpc.Pipelines[i]
+		switch pipeline.Name {
 		case "docker-build":
-			bundles.DockerBuildBundle = bundleRef
+			bundles.DockerBuildBundle = pipeline.Bundle
 		case "fbc-builder":
-			bundles.FBCBuilderBundle = bundleRef
+			bundles.FBCBuilderBundle = pipeline.Bundle
 		case "java-builder":
-			bundles.JavaBuilderBundle = bundleRef
+			bundles.JavaBuilderBundle = pipeline.Bundle
 		case "nodejs-builder":
-			bundles.NodeJSBuilderBundle = bundleRef
+			bundles.NodeJSBuilderBundle = pipeline.Bundle
```

Comment:
nit: should these cases be removed? (I'm asking because they're missing in the [config](https://raw.githubusercontent.com/redhat-appstudio/infra-deployments/main/components/build-service/base/build-pipeline-config/build-pipeline-config.yaml))

Reply:
Yes, correct. I had left them just for a case if we add them later, but we can reintroduce the fields then. Will remove.